src/test/java/com/revature/studyforce/notification/integration/FlashcardSubscriptionIntegrationTest.java

```diff
@@ -0,0 +1,121 @@
+package com.revature.studyforce.notification.integration;
+
+import com.revature.studyforce.flashcard.model.Flashcard;
+import com.revature.studyforce.flashcard.repository.FlashcardRepository;
+import com.revature.studyforce.notification.controller.SubscriptionController;
+import com.revature.studyforce.notification.model.FlashcardSubscription;
+import com.revature.studyforce.notification.model.Subscription;
+import com.revature.studyforce.notification.repository.FlashcardSubscriptionRepository;
+import com.revature.studyforce.notification.repository.SubscriptionRepository;
+import com.revature.studyforce.notification.service.FlashcardSubscriptionService;
+import com.revature.studyforce.notification.service.SubscriptionService;
+import com.revature.studyforce.user.model.Authority;
+import com.revature.studyforce.user.model.User;
+import com.revature.studyforce.user.repository.UserRepository;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+
+import java.sql.Timestamp;
+import java.time.LocalDateTime;
+
+@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD)
+@SpringBootTest
+@AutoConfigureMockMvc
+@TestPropertySource(locations = "classpath:test-application.properties")
+public class FlashcardSubscriptionIntegrationTest {
```

Comment:
@author Documentation required and @link to service layer it is testing.

Reply:
fixed